pkg/controller/job/indexed_job_utils.go

```diff
@@ -341,3 +341,47 @@ func completionModeStr(job *batch.Job) string {
 	}
 	return string(batch.NonIndexedCompletion)
 }
+
+// parseCompletedIndexes parses the interval string from
+// job.Status.CompletedIndexes and returns a map of unique
+// completed indexes.
+func parseCompletedIndexes(completedIndexes string) map[int]bool {
```

Comment:
can't you use `succeededIndexesFromJob`? Or at least refactor that function to fit the new needs.

Note that using `map[int]` is very problematic, because the number of completions can be very large. `orderedIntervals` is a compressed version of a map.

Reply:
Makes sense, I have implemented the refactor of `succeededIndexesFromJob ` and updated my approach accordingly.